mm2src/coins/lp_coins.rs

```diff
@@ -5751,3 +5755,81 @@ pub mod for_tests {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20ApproveRequest {
+    coin: String,
+    spender: EthAddress,
+    amount: BigDecimal,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20AllowanceRequest {
+    coin: String,
+    spender: EthAddress,
+}
+
+#[derive(Debug, Deserialize, Display, EnumFromStringify, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum Erc20CallError {
+    #[display(fmt = "No such coin {}", coin)]
+    NoSuchCoin { coin: String },
+    #[display(fmt = "Coin not supported {}", coin)]
+    CoinNotSupported { coin: String },
+    #[from_stringify("NumConversError")]
+    #[display(fmt = "Invalid param: {}", _0)]
+    InvalidParam(String),
+    #[from_stringify("TransactionErr")]
+    #[display(fmt = "Transaction error {}", _0)]
+    TransactionError(String),
+    #[from_stringify("Web3RpcError")]
+    #[display(fmt = "Web3 RPC error {}", _0)]
+    Web3RpcError(String),
+}
+
+impl HttpStatusCode for Erc20CallError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            Erc20CallError::NoSuchCoin { .. }
+            | Erc20CallError::CoinNotSupported { .. }
+            | Erc20CallError::InvalidParam(_)
+            | Erc20CallError::TransactionError(_)
+            | Erc20CallError::Web3RpcError(_) => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+type Erc20AllowanceResult = MmResult<BigDecimal, Erc20CallError>;
+type Erc20ApproveResult = MmResult<BytesJson, Erc20CallError>;
```

Comment:
Actual results are already quite clear, why is this needed?

Reply:
removed